lib/NeuraDialect/Mapping/mapping_util.cpp

```diff
@@ -911,50 +913,65 @@ mlir::neura::calculateAward(Operation *op, std::set<Operation *> &critical_ops,
                    backward_user_loc.time_step + mapping_state.getII());
       backward_users_locs.push_back(backward_user_loc);
     }
-    int award = 2 * mapping_state.getII();
+
+    // === Tile-based award (independent of time) ===
+    int tile_award = 0;
+
+    // Computes proximity bonus to producers. Closer tiles get higher scores.
+    int hops_to_producers = getPhysicalHops(producers, tile, mapping_state);
+    // Assumes max possible hops on a typical 4x4 grid is about 6 per producer.
+    int max_hops = static_cast<int>(producers.size()) * 6;
```

Comment:
Instead of `6`, can we check the arch.x_dim and y_dim, and make this `6` be `kMaxHops = x_dim + y_dim - 2`?